File: Missing number in matrix.java
class Solution
{
    public long  MissingNo(int[][] matrix)
    {
        // code here
        TreeSet<Long>nonzero=new TreeSet<>(), zero=new TreeSet<>();
    	
    	// Interating through rows
    	for(int i = 0; i < matrix.length; i++){
    		boolean f = true;
    		long sum = 0;
    		for(int j = 0; j < matrix[0].length; j++){
    			sum += matrix[i][j];
    			if(matrix[i][j]==0)
    				f = false;
    		}
    		if(f)nonzero.add(sum);
    		if(!f)zero.add(sum);
    	}
    	// Interating through coloumns
    	for(int  j = 0; j < matrix[0].length; j++){
    		boolean f = true;
    		long sum = 0;
    		for(int i = 0; i < matrix.length; i++){
    			sum += matrix[i][j];
    			if(matrix[i][j]==0)
    				f = false;
    		}
    		if(f)nonzero.add(sum);
    		if(!f)zero.add(sum);
    	}
    	// Interating through diagonals
    	boolean f = true;
    	long sum = 0;
    	for(int i = 0; i < matrix.length; i++){
    		sum += matrix[i][i];
    		if(matrix[i][i]==0)
    			f = false;
    	}
    	if(f)nonzero.add(sum);
		if(!f)zero.add(sum);
    	// Interating through diagonals
    	f = true;
    	sum = 0;
    	for(int i = 0, j = matrix.length - 1; j >= 0; i++,j--){
    		sum += matrix[i][j];
    		if(matrix[i][j]==0)
    			f = false;
    	}
		if(f)nonzero.add(sum);
		if(!f)zero.add(sum);
    	if(nonzero.size() == 1 && zero.size() == 1){
    		long x = nonzero.first();
    		long y = zero.first();
    		if(x <= y)
    			return -1;
    		return x - y;
    	}
    	return -1;
    }
}

File: Missing number in matrix.cpp
class Solution {
public:
    long long int MissingNo(vector<vector<int> >& matrix) {
    	set<long long int>nonzero, zero;
    	
    	// Interating through rows
    	for(int i = 0; i < matrix.size(); i++){
    		bool f = true;
    		long long int sum = 0;
    		for(int j = 0; j < matrix[0].size(); j++){
    			sum += matrix[i][j];
    			if(!matrix[i][j])
    				f = false;
    		}
    		if(f)nonzero.insert(sum);
    		if(!f)zero.insert(sum);
    	}
    	// Interating through coloumns
    	for(int  j = 0; j < matrix[0].size(); j++){
    		bool f = true;
    		long long int sum = 0;
    		for(int i = 0; i < matrix.size(); i++){
    			sum += matrix[i][j];
    			if(!matrix[i][j])
    				f = false;
    		}
    		if(f)nonzero.insert(sum);
    		if(!f)zero.insert(sum);
    	}
    	// Interating through diagonals
    	bool f = true;
    	long long int sum = 0;
    	for(int i = 0; i < matrix.size(); i++){
    		sum += matrix[i][i];
    		if(!matrix[i][i])
    			f = false;
    	}
    	if(f)nonzero.insert(sum);
		if(!f)zero.insert(sum);
    	// Interating through diagonals
    	f = true;
    	sum = 0;
    	for(int i = 0, j = matrix.size() - 1; j >= 0; i++,j--){
    		sum += matrix[i][j];
    		if(!matrix[i][j])
    			f = false;
    	}
		if(f)nonzero.insert(sum);
		if(!f)zero.insert(sum);
    	if(nonzero.size() == 1 and zero.size() == 1){
    		long long int x = *nonzero.begin();
    		long long int y = *zero.begin();
    		if(x <= y)
    			return -1;
    		return x - y;
    	}
    	return -1;
    }
};
